statistics/handle/handle.go

```diff
@@ -516,20 +525,24 @@ func (h *Handle) GetPartitionStats(tblInfo *model.TableInfo, pid int64) *statist
 	return tbl
 }
 
-func (h *Handle) updateStatsCache(newCache statsCache) {
+func (h *Handle) updateStatsCache(newCache statsCache) (updated bool) {
```

Comment:
I think we should remind other people in the comments that this function may fail when `version` is not changed and retry may be needed.